ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -365,49 +412,62 @@ def _toGA4GH(self, annotation):
             id_, ontologySource = self.namespaceSplit(location['id'])
             name = location['id']
 
-        f = protocol.Feature()
-        f.featureType = protocol.OntologyTerm.fromJsonDict({
-            "term": name,
-            "id": id_,
-            "sourceName": ontologySource})
-        f.id = annotation['id']  # TODO connect with real feature Ids
-        f.featureSetId = ''
-        f.parentIds = []
-        f.attributes = protocol.Attributes.fromJsonDict({"vals": {}})
-
-        id_, ontologySource = self.namespaceSplit(
+        f = self._generateFeature(
```

Comment:
I'm not convinced by this way of organising things (i.e, calling ``_generateFeature(..)``) with a whole bunch of parameters... There's a lot of places for things to go wrong here between where we're passing the value in, and where it eventually gets assigned into the object. It just feels a bit brittle. I've spent about 5 minutes tracing up and down the file trying to figure out what's happening, so I don't think it's that clear in any case. What problem are we trying to solve here with ``_generateFeature``?